### PR TITLE
Plane: Send a quadplane version of NAV_CONTROLLER_OUTPUT

### LINKED ISSUE #9002: MAVLink: Quadplane: QLoiter does not report altitude error correctly
### Bug report

**Issue details**

When in quadplane modes altitude error on `NAV_CONTROLLER_OUTPUT` is not updated. This is probably happening on more fields then just altitude error, but that's the one I noticed this on.

**Version**
f7f4b43683b09

**Platform**
[  ] All
[  ] AntennaTracker
[  ] Copter
[ x ] Plane
[  ] Rover
[  ] Submarine

**Airframe type**
QuadPlane

**Hardware type**
SITL

**Reproduction**
Launch quadplane SITL, run the following commands in MAVProxy:
```
module load graph
graph NAV_CONTROLLER_OUTPUT.alt_error
arm throttle
qloiter
rc 3 2000
<... wait some amount of time, observe the graph ...>
rc 3 1500
```
Even if you accept not updating the altitude target during the climb (which would be wrong) it should definitely be corrected once in a hover.

### PR DESCRIPTION
**Pull request by WickedShell**
Send a quadplane appropriate version of NAV_CONTROLLER_OUTPUT to the GCS. Resolves #9002

### CODE REVIEW

**Comment by WickedShell** [comment]
#9004 frees up the required space for the feature on px4-v2.

File: ArduPlane/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -180,16 +180,32 @@ void Plane::send_extended_status1(mavlink_channel_t chan)
...
+        bool wp_nav_valid = plane.control_mode == QLOITER || plane.control_mode == QLAND || plane.control_mode == QRTL;
+
+        mavlink_msg_nav_controller_output_send(
+            chan,
+            targets.x * 1.0e-2f,
+            targets.y * 1.0e-2f,
+            targets.z * 1.0e-2f,
+            wp_nav_valid ? quadplane.wp_nav->get_wp_bearing_to_destination() : 0,
+            wp_nav_valid ? MIN(quadplane.wp_nav->get_wp_distance_to_destination(), UINT16_MAX) : 0,
+            plane.control_mode != QSTABILIZE ? quadplane.pos_control->get_alt_error() * 1.0e-2f : 0,
```
Comment (peterbarker) [inline ArduPlane/GCS_Mavlink.cpp]: Add prens.

File: ArduPlane/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -180,16 +180,32 @@ void Plane::send_extended_status1(mavlink_channel_t chan)
...
-        nav_pitch_cd * 0.01f,
-        nav_controller->nav_bearing_cd() * 0.01f,
-        nav_controller->target_bearing_cd() * 0.01f,
-        MIN(auto_state.wp_distance, UINT16_MAX),
-        altitude_error_cm * 0.01f,
-        airspeed_error * 100,
-        nav_controller->crosstrack_error());
+    if (quadplane.in_vtol_mode()) {
+        const Vector3f &targets = quadplane.attitude_control->get_att_target_euler_cd();
+        bool wp_nav_valid = plane.control_mode == QLOITER || plane.control_mode == QLAND || plane.control_mode == QRTL;
```
Comment (peterbarker) [inline ArduPlane/GCS_Mavlink.cpp]: Maybe add a method?

Comment (WickedShell) [reply ArduPlane/GCS_Mavlink.cpp]: I wasn't confident enough if this was a dirty hack or generally sufficient to make it a method...

**Pull request merged**